openwisp_notifications/tests/test_notifications.py

```diff
@@ -152,3 +160,48 @@ def test_organization_recipient(self):
             notifications = n[1]
             for notification, recipient in zip(notifications, recipents):
                 self.assertEqual(notification.recipient, recipient)
+
+    def test_default_notification_type(self):
+        self.notification_options.pop('verb')
+        self.notification_options.update({'type': 'default'})
+        self._create_notification()
+        n = notification_queryset.first()
+        self.assertEqual(n.level, 'info')
+        self.assertEqual(n.verb, 'default verb')
+
+    def test_message_template(self):
+        self.notification_options.pop('description')
+        self._create_notification()
+        n = notification_queryset.first()
+        self.notification_options.update({'actor': self.admin})
+        exp_desc = render_to_string(
+            MESSAGE_TEMPLATE, self.notification_options
+        ).rstrip()
+        self.assertEqual(n.description, exp_desc)
+
+    def test_register_unregister_notification_type(self):
+        custom_type = {
+            'test type': {
+                'name': 'Test Notification Type',
+                'level': 'test',
+                'verb': 'testing',
+            }
+        }
+        # Registering new notification type
+        register_notification_type(custom_type)
+        self.notification_options.update({'type': 'test type'})
+        self._create_notification()
+        n = notification_queryset.first()
+        self.assertEqual(n.level, 'test')
+        self.assertEqual(n.verb, 'testing')
+
+        # Unregistering a notification type which does not exists
+        with self.assertRaises(ImproperlyConfigured):
+            unregister_notifcation_type('wrong type')
+
+        # Unregistering notification type it's name
+        unregister_notifcation_type('test type')
+        self._create_notification()
+        n = notification_queryset.first()
+        self.assertEqual(n.level, 'info')
+        self.assertEqual(n.verb, 'Test Notification')
```

Comment:
since this test is pretty long, maybe we can use subtests, it should help: https://www.caktusgroup.com/blog/2017/05/29/subtests-are-best/

We should start using more of this, me incldued. This is a good time to start.

@PabloCastellano @nepython @NoumbissiValere @atb00ker take note of this :point_up:

Reply:
Subtests looks cool, I will rewrite this implementation. :smile: